### PR TITLE
Verify hard disk size before creating lvm

### PR DESCRIPTION
**Pull request by waynechen55**
* **Introduce** new subroutine get_hard_disk_size to verify hard disk size is equal to or greater than 20G before creating lvm with it. 

* **This** can help ensure test run proceed with suitable hard disk and increase test run stability, otherwise test fail may fail like [this](https://openqa.suse.de/tests/13348898#step/libvirt_extend_storage_lvm/62).

* **If** there is no suitable disk to be used, fail test run immediately and give explicit error message. (At the moment, all SUTs have at least on spare disk size of which is more than 20G, so test run will not fail. ```die``` only serves the purpose of raising alert in the future if certain machine does not have qualified disk, so it can be excluded from taking such test run by modifying worker and job yaml files).

* **Verification Runs:**
  * [verification run 01 on 12sp5](https://openqa.suse.de/tests/13571904)
  * [verification run 02 on 12sp5](https://openqa.suse.de/tests/13571906)
  * [verification run 03 on 15sp6](https://openqa.suse.de/tests/13572010)
  * [verification run 04 on 15sp6](https://openqa.suse.de/tests/13572011)
  * [verification run 05 on 15sp6](https://openqa.suse.de/tests/13572012)
  * [verification run 06 frequently failed imagetester:4 also passed](https://openqa.suse.de/tests/13564729)
  * [verification run 07 failed example on 12sp5](https://openqa.suse.de/tests/13579900)
  * [verification run 08 failed example on 15sp6](https://openqa.suse.de/tests/13579901)

### CODE REVIEW

**Comment by waynechen55** [comment]
@alice-suse @guoxuguang @Julie-CAO @RoyCai7 @nanzhg @tbaev Please have a look.

**Comment by Julie-CAO** [comment]
I remember there are 240Gx3 and 560Gx1 storage on Kermit, is it not enough for extra 20G storage for lvm test? what requirement is for lvm test on storage? Could we fulfill this in host autoyast?

Because I think "die" and aborting test is not a good idea. More applicable ways:

1. run lvm storage tests on machines which can affort the storage requirements.
2. continue subsequent feature tests even if there is not enough storage.

**Comment by Julie-CAO** [comment]
I remember there are 240Gx3 and 560Gx1 storage on Kermit, is it not enough for extra 20G storage for lvm test? what requirement is for lvm test on storage? Could we fulfill this in host autoyast?

Because I think "die" and aborting test is not a good idea. More applicable ways:

1. run lvm storage tests on machines which can affort the storage requirements.
2. continue subsequent feature tests even if there is not enough storage.

**Comment by Julie-CAO** [comment]
I remember there are 240Gx3 and 560Gx1 storage on Kermit, is it not enough for extra 20G storage for lvm test? what requirement is for lvm test on storage? Could we fulfill this in host autoyast?

Because I think "die" and aborting test is not a good idea. More applicable ways:

1. run lvm storage tests on machines which can affort the storage requirements.
2. continue subsequent feature tests even if there is not enough storage.

**Comment by waynechen55** [comment]
> I remember there are 240Gx3 and 560Gx1 storage on Kermit, is it not enough for extra 20G storage for lvm test? what requirement is for lvm test on storage? Could we fulfill this in host autoyast?
> 
> Because I think "die" and aborting test is not a good idea. More applicable ways:
> 
>     1. run lvm storage tests on machines which can affort the storage requirements.
> 
>     2. continue subsequent feature tests even if there is not enough storage.

Not all disks are more than 20G, it might be sda, sdb, sdc or even sdd.
Test run failed from time to time because the order in which disks are stored in perl list is random, so test may choose unqualified disk. Verifying disk size more than 20G is enough. All machines, at least, have a disk more than 20G, so current implementation is enough. If, in the future, test run really failed due to there is no such disk, I can update worker/yaml files to let test only run with qualified machine. I do not think there is a need to do this at the moment.

File: tests/virt_autotest/libvirt_extend_storage_lvm.pm [hunk 1]
```diff
@@ -69,18 +69,32 @@ sub prepare_lvm_storage_pool_source {
...
+    else {
+        die("Failed to find spare hard disk with 20G or more size");
+    }
+}
+
+#Get disk size in gigabytes
+sub get_hard_disk_size {
+    my $hard_disk_name = shift;
+
+    my $hard_disk_size = script_output("lsblk /dev/$hard_disk_name -n -l -b -o SIZE -d -e 7,11");
```
[the remark this answers is not in the document]
Comment (waynechen55) [reply tests/virt_autotest/libvirt_extend_storage_lvm.pm]: Done.

**Comment by Julie-CAO** [comment]
I thought if the choosen disk is less than 20G than die, now I get the point than only othose disks with size greater than 20G can be choosen. It is good.

**Comment by alice-suse** [comment]
LGTM

**Comment by alice-suse** [comment]
Do I need to wait for new validation before merge?

**Comment by waynechen55** [comment]
> Do I need to wait for new validation before merge?

Ready for merge.

**Pull request merged**